tools/check_typehints.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python
```

Comment:
What is this script used for?

Reply:
Added a docstring. 

Short version is custom checking that since I figured out configuring `stubtest` are _mostly_ not needed, but will keep around for at least the initial roll out stages as an additional (manual, not CI) test to run as a sanity check/for things stubtest currently ignores.